e2e/testcases/apiservice_test.go

```diff
@@ -17,36 +17,114 @@ package e2e
 	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/ntopts"
 	"kpt.dev/configsync/pkg/api/configsync"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+)
+
+const (
+	adapterNamespace = "custom-metrics"
+	adapterName      = "custom-metrics-stackdriver-adapter"
 )
 
 func TestCreateAPIServiceAndEndpointInTheSameCommit(t *testing.T) {
-	nt := nomostest.New(t)
+	nt := nomostest.New(t, ntopts.Unstructured, ntopts.RequireGKE(t))
+	t.Cleanup(func() {
+		if t.Failed() {
+			nt.PodLogs(adapterNamespace, adapterName, "pod-custom-metrics-stackdriver-adapter", true)
+		}
+	})
 	nt.T.Log("Creating commit with APIService and Deployment")
+	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/namespace.yaml", "acme/namespaces/custom-metrics/namespace.yaml")
 	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/namespace-custom-metrics.yaml", "acme/namespaces/custom-metrics/namespace-custom-metrics.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/rbac.yaml", "acme/cluster/rbac.yaml")
 	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/apiservice.yaml", "acme/cluster/apiservice.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush("add testing resources")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("adding apiservice resources")
 	nt.T.Log("Waiting for nomos to sync new APIService")
 	nt.WaitForRepoSyncs()
 
-	nt.T.Log("Removing APIService")
+	err := validateStackdriverAdapterStatusCurrent(nt)
+	if err != nil {
+		nt.T.Fatalf("validate failed %s", err)
+	}
+
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/cluster/apiservice.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove API service")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter API service")
+	nt.WaitForRepoSyncs()
+
+	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace-custom-metrics.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter namespace")
 	nt.WaitForRepoSyncs()
 }
 
-func TestImporterAndSyncerResilientToBadAPIService(t *testing.T) {
-	nt := nomostest.New(t)
-	nt.T.Log("Adding bad APIService")
+func TestImporterAndSyncerResilientToFlakyAPIService(t *testing.T) {
+	nt := nomostest.New(t, ntopts.RequireGKE(t), ntopts.Unstructured)
+	nt.T.Log("Adding API service with backend temporarily unavailable")
 	nt.MustKubectl("apply", "-f", "../testdata/apiservice/apiservice.yaml")
 	nt.T.Cleanup(func() {
 		nt.MustKubectl("delete", "-f", "../testdata/apiservice/apiservice.yaml", "--ignore-not-found")
+		if t.Failed() {
+			nt.PodLogs(adapterNamespace, adapterName, "pod-custom-metrics-stackdriver-adapter", true)
```

Comment:
Is this pod one of the ones in `apiservice.yaml`? If so, print he pod logs before deleting.